internal/kafka/kafka.go

```diff
@@ -206,25 +206,29 @@ func (b *kafkaBroker) Consume(ctx context.Context, topic string, since time.Time
 				msg.Key, msg.Topic, msg.Offset, msg.Partition)
 
 			// build new context - identity and trace id
-			ctx, err = identity.WithIdentityFrom64(ctx, header("x-rh-identity", msg.Headers))
-			if err != nil {
-				logger.Trace().Msgf("Could not extract identity from context to Kafka message: %s", err)
+			newLogger := logger.With().Str("msg_id", random.TraceID().String())
```

Comment:
The idea behind this is to have at least some unique identifier just in case there is no identity in the incoming message (thus no span - no trace id). It will be used in the "else" block down below so we can at least correlate logs from the same kafka (anonymous) message.